### DISCUSSION TITLE
Prevent useFetch from automatically executing when query param ref changes

### DISCUSSION THREAD

**Discussion by chrislee973** [nuxt/nuxt#18440] [Questions] [answered]
When the query param is a ref, and the param ref changes, is there a way to prevent the fetch from occurring automatically and instead only fetch when the execute() function is called? 
```javascript
const count = ref(1);
// The api is called every time the count ref changes, but I only want it to be called when execute() function is run.
const { execute, data } = await useFetch(
  () => `/api/hello?count=${count.value}`,
  {
    immediate: false,
    watch: [count],
  }
);
```

See stackblitz: https://stackblitz.com/edit/github-qpqckm?file=app.vue,server%2Fapi%2Fhello.ts

Is there a way to only fetch when the "Fetch" button is clicked and not when the "count" ref is incremented?

**Comment by oritwoen** [comment, marked as answer]
For such actions, `useAsyncData` is more suitable, which by default does not observe parameters. Below I give you a corrected example in the form of code and a working online demo.

Online demo: https://stackblitz.com/edit/github-qpqckm-bqhqv9?file=app.vue,server%2Fapi%2Fhello.ts

```vue
<script setup>
const count = ref(1);

const { data, refresh } = await useAsyncData('counter', async () => {
  return await $fetch(`/api/hello?count=${count.value}`);
});
</script>

<template>
  <NuxtExampleLayout example="composables/use-fetch">
    <div>
      <div>count: {{ count }}</div>
      <NButton @click="count++"> Increment count </NButton>
      <div></div>
      <NButton @click="refresh()">Fetch</NButton>
      <div>Data:</div>
      <pre class="text-left"><code>{{ data }}</code></pre>
    </div>
  </NuxtExampleLayout>
</template>
```

**Reply by chrislee973** [reply]
Thanks! This works
